aikau/src/main/resources/alfresco/lists/Paginator.js

```diff
@@ -221,21 +239,45 @@ define(["dojo/_base/declare",
        */
       processLoadedDocuments: function alfresco_lists_Paginator__processLoadedDocuments(payload) {
          // jshint maxcomplexity:false,unused:false,maxstatements:false
-         if (payload && payload.totalDocuments !== null && payload.startIndex !== null)
+         var totalRecords = lang.getObject(this.totalResultsProperty, false, payload);
+         var startIndex = lang.getObject(this.startIndexProperty, false, payload);
+         if (payload && totalRecords !== null && startIndex !== null)
          {
```

Comment:
If lang.getObject fails to find a result, I think it returns undefined, not null. Probably worth you confirming my findings before you change this!